File: leetcode/0049-group-anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        """
        RUNTIME: 99 ms (96.5%)
        MEMORY: 17.1 MB (95.96%)
        """
        hashmap = {}
        for i in strs:
            tmp = ''.join(sorted(list(i)))
            if tmp not in hashmap:
                hashmap[tmp] = [i]
            else:
                hashmap[tmp].append(i)
        return list(hashmap.values())

File: leetcode/0049-group-anagrams/solution.cpp
class Solution {
public:
    /**
     * RUNTIME: 39 ms (91.77%)
     * MEMORY: 20.8 MB (40.23%)
    */
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map <string, vector<string>> hashmap;
        for (string i : strs) {
            string tmp = i;
            sort(tmp.begin(), tmp.end());
            if (hashmap.find(tmp) == hashmap.end()) {
                hashmap[tmp] = vector<string>();
            }
            hashmap[tmp].push_back(i);
        }
        vector<vector<string>> result;
        for (auto i : hashmap) {
            result.push_back(i.second);
        }
        return result;
    }
};
